### DISCUSSION TITLE
Windows-native-notifications new in Firefox V.111.0

### DISCUSSION THREAD

**Discussion by Mautzimann** [Cookie-AutoDelete/Cookie-AutoDelete#1503] [Q&A / Support]
I wouldnt create a new issue, but I saw something, since Firefox is now using windows native notifications:
- some of CAD notifications will not be kept in the time of seconds (5 sec in my case), they disappear very fast
- maybe some text in the "toast" is truncated?
Is this a restriction of notification API of Firefox?

**Comment by Mautzimann** [comment]
If restrictions dont allow the 2 fixes / improvements, maybe it is possible in the log of deleted cookies etc. to log also manually deletions?

**Comment by kennethtran93** [comment]
> I wouldnt create a new issue, but I saw something, since Firefox is now using windows native notifications:
> 
>     * some of CAD notifications will not be kept in the time of seconds (5 sec in my case), they disappear very fast
> 
>     * maybe some text in the "toast" is truncated?
>       Is this a restriction of notification API of Firefox?

We do not have the capability to determine if a Firefox notification or Native notification that is used.  As for truncation, there doesn't seem to be any limit specified in the API itself, so it may be a native limit which I cannot determine.

In other words, yes api limitations.

> If restrictions dont allow the 2 fixes / improvements, maybe it is possible in the log of deleted cookies etc. to log also manually deletions?

Possibly - if there's no feature request for creating a log entry for manual deletions, please create one.

From the existing code, there _could_ be a small adjustment to the notification duration code in that I was starting the timer the moment the create function was executed, which may or may not actually show the notification immediately.  As this function is using promises, I could change it to only start the timer once the promise resolves itself, with a caveat noted here:
> [Promise] fulfilled when the notification is created and the display process has been started, which is before the notification is actually displayed to the user.

However, there's also a note on the notification api page about trying to create more than one notification in a short amount of time
>  If you call notifications.create() more than once in rapid succession, Firefox may end up not displaying any notification at all.

A potential workaround for this would be to only create notifications when the existing one is closed.  However, doing so may potentially queue/delay future notifications if the existing one is not cleared.

Another workaround would be to only create a notification (if back to back) about 1-2 seconds after the last one was created.

Another workaround would be to not even clear the notifications at all...as an option.

**Comment by LaurentGrenet** [comment]
Hi @kennethtran93 ,

there is another issue linked to this Firefox evolution : it is the fact that now notifications issued from CAD are sounding a beep... that is very quickly very annoying, since in case of frequent domain change when browsing, there are a lot and a lot of such beeps.

Of course, it is feasible to ask in W10 parameters that Firefox notifications are all and always silent.

But I'm wondering if there is some parameter in the api allowing CAD to send its notification requesting for them a silent mode, whatever the global parameter for Firefox's notifications.
This would allow to let all other (that are actually "not expected", at least less than CAD's ones) notification beeping, and not be disturbed by beep of CAD's notification (that are on the contrary "expected")

Thank you for advice... and implementation is feasible.

**Reply by kennethtran93** [reply]
Sorry, but there's no parameter for silent notifications available. (https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/notifications/NotificationOptions).

**Comment by LaurentGrenet** [comment]
Thank you for answering.
Not a very important problem. I was only asking in case it could be available.
I have stopped sound for all firefox notifications.
